sherpa-onnx/csrc/sherpa-onnx-offline-parallel.cc

```diff
@@ -0,0 +1,292 @@
+// sherpa-onnx/csrc/sherpa-onnx-offline.cc
+//
+// Copyright (c)  2022-2023  Xiaomi Corporation
+
+#include <stdio.h>
+
+#include <atomic>
+#include <chrono>  // NOLINT
+#include <fstream>
+#include <mutex>  // NOLINT
+#include <string>
+#include <thread>  // NOLINT
+#include <vector>
+
+#include "sherpa-onnx/csrc/offline-recognizer.h"
+#include "sherpa-onnx/csrc/parse-options.h"
+#include "sherpa-onnx/csrc/wave-reader.h"
+
+std::atomic<int> wav_index(0);
+std::mutex mtx;
+
+std::vector<std::vector<std::string>> split_to_batchsize(
+    const std::vector<std::string> &input, int32_t batch_size) {
+  std::vector<std::vector<std::string>> outputs;
+  auto itr = input.cbegin();
+  int32_t process_num = 0;
+
+  while (process_num + batch_size <= static_cast<int32_t>(input.size())) {
+    auto chunk_end = itr + batch_size;
+    outputs.emplace_back(itr, chunk_end);
+    itr = chunk_end;
+    process_num += batch_size;
+  }
+  if (itr != input.cend()) {
+     outputs.emplace_back(itr, input.cend());
+  }
+  return outputs;
+}
+
+std::vector<std::string> load_scp_file(const std::string &wav_scp_path) {
+  std::vector<std::string> wav_paths;
+  std::ifstream in(wav_scp_path);
+  if (!in.is_open()) {
+      fprintf(stderr, "Failed to open file: %s.\n", wav_scp_path.c_str());
+      return wav_paths;
+  }
+  std::string line, column1, column2;
+  while (std::getline(in, line)) {
+    std::istringstream iss(line);
+    iss >> column1 >> column2;
+    wav_paths.emplace_back(std::move(column2));
+  }
+  in.close();
+
+  return wav_paths;
+}
+
+void asr_inference(const std::vector<std::vector<std::string>> &chunk_wav_paths,
+                   sherpa_onnx::OfflineRecognizer* recognizer,
+                   float* total_length, float* total_time) {
+  std::vector<std::unique_ptr<sherpa_onnx::OfflineStream>> ss;
+  std::vector<sherpa_onnx::OfflineStream *> ss_pointers;
+  float duration = 0.0f;
+  float elapsed_seconds_batch = 0.0f;
+
+  // warm up
+  for (const auto &wav_filename : chunk_wav_paths[0]) {
+    int32_t sampling_rate = -1;
+    bool is_ok = false;
+    const std::vector<float> samples =
+        sherpa_onnx::ReadWave(wav_filename, &sampling_rate, &is_ok);
+    if (!is_ok) {
+      fprintf(stderr, "Failed to read %s\n", wav_filename.c_str());
```

Comment:
```suggestion
      fprintf(stderr, "Failed to read %s\n", wav_filename.c_str());
      continue;
```

Use `continue` to skip this wave.